feat: implement summary tag component

briansztamfater opened the pull request:
fixes #17848 

### Summary

Implements Summary tag component

https://github.com/status-im/status-mobile/assets/18485527/03da09dc-2c92-44cb-9692-05877910d949

#### Platforms

- Android
- iOS

### Steps to test

- Open Status
- Open quo preview
- Select tag > Summary tag
- Test component

status: ready

briansztamfater committed:
feat: implement summary tag component

Signed-off-by: Brian Sztamfater <brian@status.im>

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 5c539dcf | [#2](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-17858/2/) | 2023-11-08 23:40:05 | ~5 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/4xiT2B) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fi.diawi.com%2F4xiT2B)|

J-Son89 commented on src/quo/components/tags/summary_tag/view.cljs:
```diff
@@ -0,0 +1,80 @@
+                              :address       address
+                              "")
+        customization-color (case type
+                              :account       (:customization-color account-props)
```
imo we should pass `customization-color` as it's own prop instead 👍 

i.e 
```
(defn- view-internal
  [{:keys [customization-color ...
           address type
           theme]
           ```

J-Son89 commented on src/quo/components/tags/summary_tag/view.cljs:
```diff
@@ -0,0 +1,80 @@
+           address type
+           theme]
+    :as   props}]
+  (let [text                (case type
```
I think we're making the api of this component really confusing here. Why not just have a "text" param and handle this outside in application code.  although I would probably change "text" to something like "label"

ulisesmac replied:
I agree too, I think this component should only receive the params it needs to work. Right now it's hard to know what keys are expected in those maps (account-props, user-props, network-props, etc) to make this work.

Also a docstring specifying the props supported for each type could work :+1:

J-Son89 commented on src/quo/components/tags/summary_tag/view.cljs:
```diff
@@ -0,0 +1,80 @@
+     {:source (resources/get-token (keyword (string/lower-case (:symbol token-props))))
+      :style  {:width  24
+               :height 24}}]
+    nil))
```
I personally would prefer this as something like:
```
(defn- left-view
  [{:keys [type  emoji customization-color label image-source
           ]}]
  (case type
    :account
    [account-avatar/view
     {:customization-color customization-color
      :size                24
      :emoji               emoji
      :type                :default}]
    :network
    [rn/image
     {:source image-source
      :style  {:width  24
               :height 24}}]
    :saved-address
    [wallet-user-avatar/wallet-user-avatar
     {:full-name           label
      :size                :size-24
      :customization-color customization-color }]
    :collectible
    [rn/image
     {:source image-source
      :style  style/collectible-image}]
    :user
    [user-avatar/user-avatar
     {:full-name           label
      :size                :xs
      :profile-picture     image-source
      :customization-color customization-color}]
    :token
    [rn/image
     {:source image-source
      :style  {:width  24
               :height 24}}]
    nil))
 ```
 and also the `token` image resolving I prefer to do in the application code, as perhaps in some case they are urls and others based on images that we already have in the codebase.
 
 cc @ulisesmac

ulisesmac replied:
Yep, I agree. The API looks cleaner :+1:

ulisesmac commented on src/quo/components/tags/summary_tag/view.cljs:
```diff
@@ -0,0 +1,80 @@
+    :network
+    [rn/image
+     {:source (:network-logo network-props)
+      :style  {:width  24
+               :height 24}}]
```
This could be in the style ns :)

briansztamfater commented:
@ulisesmac @J-Son89 thanks for the feedback, will refactor and simplify the component API :)

vkjr commented on src/quo/components/tags/summary_tag/component_spec.cljs:
```diff
@@ -0,0 +1,51 @@
+    [quo.components.tags.summary-tag.view :as summary-tag]
+    [test-helpers.component :as h]))
+
+(h/describe "Summary tag component tests"
```
I'd suggest adding one more test where the component instantiated with empty props, to make sure it doesn't crash

vkjr replied:
Similarly, since we have a set of props packed under `:token-props`, `:account-props`, their empty state also worth testing.

❗  Right now component crashes the app in following scenarios:
```clojure
[quo/summary-tag
         {:type        :token
          :token-props {}}]
```
this also:
```clojure
[quo/summary-tag
         {:type        :account
          :account-props {}}]
```          
          
Didn't test the rest

vkjr reviewed (changes requested):
@briansztamfater, thanks for the PR!
Could you please also add tests to cover the cases when empty `:token-props`, `:account-props`, etc... passed to the component and make sure the component doesn't crash? 
Thanks!

briansztamfater commented:
> @briansztamfater, thanks for the PR! Could you please also add tests to cover the cases when empty `:token-props`, `:account-props`, etc... passed to the component and make sure the component doesn't crash? Thanks!

@vkjr Thanks for the feedback, I simplified the API so there is no token-props, account-props, etc anymore. Also refactored tests and added a case where we pass no props so we make sure it doesn't crash.

briansztamfater commented:
@Francesca-G can you please review this component? Thanks in advance!

Francesca-G reviewed (approved):
Looks good :)
Here's the [review](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3415%3A214763&mode=design&t=RdhOk7x2IfbSaoue-1) with some minor issues to check

briansztamfater commented:
@Francesca-G thanks for the feedback, updated the text in preview for the token variant (it is just a bug in the preview) and checked the height that is actually 32 points

![](https://github.com/status-im/status-mobile/assets/18485527/0bc87376-9f1e-4d37-bf9a-856cad9525c3)

Proceeding to merge, thanks!
